scipy/stats/_cdf_distance_1d.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+
+#pythran export _cdf_distance_1d(int, float[:], float[:])
+#pythran export _cdf_distance_1d(int, float[:], float[:], None, None)
+#pythran export _cdf_distance_1d(int, float[:], float[:], None, float[:])
+#pythran export _cdf_distance_1d(int, float[:], float[:], float[:], None)
+#pythran export _cdf_distance_1d(int, float[:], float[:], float[:], float[:])
+def _cdf_distance_1d(p, u_values, v_values, u_weights=None, v_weights=None):
+    r"""
+    Compute, between two one-dimensional distributions :math:`u` and
+    :math:`v`, whose respective CDFs are :math:`U` and :math:`V`, the
+    statistical distance that is defined as:
+
+    .. math::
+
+        l_p(u, v) = \left( \int_{-\infty}^{+\infty} |U-V|^p \right)^{1/p}
+
+    p is a positive parameter; p = 1 gives the Wasserstein distance, p = 2
+    gives the energy distance.
+
+    Parameters
+    ----------
+    u_values, v_values : array_like
+        Values observed in the (empirical) distribution.
+    u_weights, v_weights : array_like, optional
+        Weight for each value. If unspecified, each value is assigned the same
+        weight.
+        `u_weights` (resp. `v_weights`) must have the same length as
+        `u_values` (resp. `v_values`). If the weight sum differs from 1, it
+        must still be positive and finite so that the weights can be normalized
+        to sum to 1.
+
+    Returns
+    -------
+    distance : float
+        The computed distance between the distributions.
+
+    Notes
+    -----
+    The input distributions can be empirical, therefore coming from samples
+    whose values are effectively inputs of the function, or they can be seen as
+    generalized functions, in which case they are weighted sums of Dirac delta
+    functions located at the specified values.
+
+    References
+    ----------
+    .. [1] Bellemare, Danihelka, Dabney, Mohamed, Lakshminarayanan, Hoyer,
+           Munos "The Cramer Distance as a Solution to Biased Wasserstein
+           Gradients" (2017). :arXiv:`1705.10743`.
+
+    """
+
+    u_sorter = np.argsort(u_values)
+    v_sorter = np.argsort(v_values)
+
+    all_values = np.concatenate((u_values, v_values))
+    all_values = np.sort(all_values, kind='mergesort')
+
+    # Compute the differences between pairs of successive values of u and v.
+    deltas = np.diff(all_values)
+
+    # Get the respective positions of the values of u and v among the values of
+    # both distributions.
+    u_cdf_indices = np.searchsorted(u_values[u_sorter],all_values[:-1],'right')
+    v_cdf_indices = np.searchsorted(v_values[v_sorter],all_values[:-1],'right')
```

Comment:
Minor: if you're touching this code anyway, can you add some PEP 8 spaces after the commas?

Reply:
Sorry, I forgot that before. Just added the spaces.